contracts/test/MockERC4626.sol

```diff
@@ -207,4 +211,27 @@ contract MockERC4626 is ERC4626, MultiRolesAuthority {
         );
         return accrued;
     }
+
+    function _getAccruedInterest(uint256 timestamp) internal view returns (uint256) {
+        // If the rate is zero, no interest has accrued.
+        if (_rate == 0) {
+            return 0;
+        }
+
+        // If the block timestamp is less than last updated, the accrual
+        // calculation will underflow. This can occur when using anvil's state
+        // snapshots.
+        if (block.timestamp < _lastUpdated) {
+            return 0;
+        }
+
+        // base_balance = base_balance * (1 + r * t)
+        uint256 timeElapsed = (timestamp - _lastUpdated).divDown(
+            365 days
+        );
+        uint256 accrued = asset.balanceOf(address(this)).mulDown(
+            _rate.mulDown(timeElapsed)
+        );
```

Comment:
Timestamp based accounting creates [challenges for the frontend](https://github.com/delvtech/hyperdrive-frontend/pull/1104) which requires [targeted patches](https://github.com/delvtech/hyperdrive-frontend/blob/main/packages/hyperdrive-js-core/src/hyperdrive/steth/ReadStEthHyperdrive.ts#L127-L134). Can we try a configured update interval?

Reply:
This doesn't add any functionality that doesn't already exist, it just allows us to see into the future so that rust tests can be more exact

Reply:
Or rather I should say this doesn't change any functionality, just adds an overloaded version of what's already there for rust tests.